crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -227,6 +228,12 @@ pub(crate) async fn bench(
     )
     .await?;
 
+    while client.writen_piece_count() < write_pieces_size / PIECE_SIZE as u64 {
+        tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
I'm not a fan of `sleep`s, usually indication of other issues. I think moving code from `BenchRpcClient` out would allow you to wait for precise number of pieces you want to write instead of doing hacks like this.

Reply:
It is removed in the following prs. This is just a stub for now.

Reply:
All sleeps removed in following PRs? I've seen some in the RPC client too.

Reply:
Yeah, there is one in the bench rpc client, but it is for salt changing, which I decided to make time relative. Should I make it segment or piece relative?

Reply:
Yes, we should avoid timeouts in these things such that they finish as soon as they can and don't depend on machine speed (making tests/benches hard to reproduce).